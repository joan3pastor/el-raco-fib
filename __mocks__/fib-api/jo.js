export const jo = {
    "assignatures": "https://api.fib.upc.edu/v2/jo/assignatures/",
    "avisos": "https://api.fib.upc.edu/v2/jo/avisos/",
    "classes": "https://api.fib.upc.edu/v2/jo/classes/",
    "foto": "https://api.fib.upc.edu/v2/jo/foto.jpg",
    "practiques": "https://api.fib.upc.edu/v2/jo/practiques/",
    "projectes": "https://api.fib.upc.edu/v2/jo/projectes/",
    "username": "joan.marc.pastor",
    "nom": "Joan Marc",
    "cognoms": "Pastor Mackenzie",
    "email": "joan.marc.pastor@est.fib.upc.edu"
}



