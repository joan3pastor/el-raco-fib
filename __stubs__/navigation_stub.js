import sinon from 'sinon';

export const navigation = {
    navigate: (screen) => {
        //console.log(`Navigation Stub: Navigated to ${screen}`);
        return screen;
    } 
}
