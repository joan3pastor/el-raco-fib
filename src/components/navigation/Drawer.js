import React, { Component } from 'react';
import { Text, ScrollView, View, TouchableOpacity, Image, ImageBackground } from 'react-native';
import { connect } from 'react-redux';
import _ from 'lodash';
import axios from 'axios';
import IconFA from 'react-native-vector-icons/FontAwesome';
import IconE from 'react-native-vector-icons/Entypo';

import { fotoURL, joURL } from '../../external_links';

export class Drawer extends Component {

  constructor(props) {
    super(props);
    this.state = { jo:{} };
  }

  fetching = false;

  me() {
    return (
      <ImageBackground source={require('../../../static/fondo2.png')} style={{height:150, marginBottom:20}}>

        <View style={{margin:12, marginBottom:9, marginTop:18, height:80, width:80, borderRadius:40, overflow:'hidden', elevation:8}}>
        {!_.isUndefined(this.props.token.token) ? <Image 
            source={{uri: fotoURL, headers: {
              'Authorization': `Bearer ${this.props.token.token}`, 
              'Accept-Language': 'ca'
            }}}
            style={{height:80, width:80, borderRadius:40, resizeMode:'cover'}}
            /> : <View />}
        </View>

        {!_.isUndefined(this.state.jo.nom) ? <View style={{marginLeft:15, flexDirection:'column'}}>
          <Text numberOfLines={1} ellipsizeMode="tail" style={{color:'#505050', fontWeight:'400'}}>{`${this.state.jo.nom} ${this.state.jo.cognoms}       `}</Text>
          <Text>{`${this.state.jo.email}       `}</Text>
        </View> : <View />}

      </ImageBackground>
    );
  }

  async getUserData() {
    if (_.isUndefined(this.state.jo.nom) && this.fetching == false) {
      this.fetching = true;
      axios(joURL, {
        headers: {'Authorization': `Bearer ${this.props.token.token}`, 'Accept-Language':'ca'}
      }).then((resp) => {
        this.fetching = false; 
        this.setState({ jo:resp.data });
      });
    }
  }

  render() {
    this.getUserData();
    return (
      <View style={{flex:1}}>
      <View>
        {this.me()}
        <ScrollView >
          <View style={{marginLeft:15}}>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('EventsDrawer')}>
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconFA name='calendar' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{` ${this.props.lang.events}      `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('NotesDrawer')}>
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconFA name='mortar-board' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{`${this.props.lang.notes}     `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('LabsDrawer')}>
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconFA name='desktop' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{`${this.props.lang.labs}     `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('SalesBiblDrawer')}>
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconFA name='book' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{` ${this.props.lang.bibl}     `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('AulesDrawer')}>
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconFA name='calendar-o' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{` ${this.props.lang.aules}     `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('PlacesDrawer')}>
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconFA name='eye' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{` ${this.props.lang.places}     `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('MobilitatDrawer')}>
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconFA name='plane' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{` ${this.props.lang.mobilitat}     `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('LecturesDrawer')}>
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconE name='blackboard' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{` ${this.props.lang.lectures}     `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('ExternalLinksDrawer')}> 
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconFA name='external-link' size={25} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{`${this.props.lang.links}     `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('FeedbackDrawer')}>
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconE name='bug' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{`${this.props.lang.feedback}     `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('SettingsDrawer')}> 
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconFA name='gear' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{`${this.props.lang.settings}     `}</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => this.props.navigation.navigate('LogOut')}>
              <View style={{alignItems:'center', height:40, flexDirection:'row'}}>
                <IconE name='log-out' size={26} color='#505050' />
                <Text style={{marginLeft:10, color:'#606060'}}>{`${this.props.lang.logout}     `}</Text>
              </View>
            </TouchableOpacity>
          </View>
        </ScrollView>
        </View>
        <View style={{flex:1}}>
          <Text style={{position:'absolute', bottom:10, right:20}}>{"v1.5. "}</Text>
        </View>
      </View>
    );
  }
}
  
function mapStateToProps(state) {
  return { 
    token: state.token,
    lang: state.lang.drawer,
  };
}
  
export default connect(mapStateToProps, {  })(Drawer);


// Nota: preguntar si .../places/ sirve para el proposito de la pesta√±a matricula ya que la fecha de actualiz. es muy antigua...

//Falta:
// - Prematricula
