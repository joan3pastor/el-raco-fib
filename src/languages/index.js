import {ca} from "./ca.js";
import {es} from "./es.js";
import {en} from "./en.js";

export default {
    ca: ca,
    es: es,
    en: en,
};