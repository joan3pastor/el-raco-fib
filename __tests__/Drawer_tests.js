import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import AxiosMockAdapter from 'axios-mock-adapter';
import _ from 'lodash';

// LANG
import {ca} from "../src/languages/ca";
const lang = ca.drawer;

import {Drawer} from "../src/components/navigation/Drawer";
import {jo} from "../__mocks__/fib-api/jo";
import {navigation} from "../__stubs__/navigation_stub";

Enzyme.configure({ adapter: new Adapter() });

beforeEach( () => {})

const wrapper = shallow(
  <Drawer token={{token:undefined}} lang={lang} />
);
const instance = wrapper.instance();

describe('Testing Drawer component', () => {
  it('renders as expected when token not yet provided', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('calls getUserData() once when token provided', () => {
    const spy_getUserData = sinon.spy(instance, "getUserData");
    wrapper.setProps({token:{token:"aToken"}, lang:lang});
    expect(spy_getUserData.calledOnce).toBe(true);
  });

  it('renders as expected when token provided', () => {
    expect(wrapper).toMatchSnapshot();
  });  

  it('renders user info correctly after fetching user data', () => {
    instance.fetching = false;
    wrapper.setState({jo: jo});
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('navigates correctly to EventsDrawer', () => {
    const navigation_spy = sinon.spy(navigation, "navigate");
    wrapper.setProps({navigation: navigation, lang:lang});
    navigation.navigate("EventsDrawer"); //todo
    expect(navigation_spy.returnValues[navigation_spy.returnValues.length-1]).toBe("EventsDrawer");
  });
  
});
