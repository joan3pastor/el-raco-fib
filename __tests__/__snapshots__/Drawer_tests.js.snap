// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Drawer component renders as expected when token not yet provided 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <ImageBackground
      source={
        Object {
          "testUri": "../../../static/fondo2.png",
        }
      }
      style={
        Object {
          "height": 150,
          "marginBottom": 20,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 40,
            "elevation": 8,
            "height": 80,
            "margin": 12,
            "marginBottom": 9,
            "marginTop": 18,
            "overflow": "hidden",
            "width": 80,
          }
        }
      >
        <View />
      </View>
      <View />
    </ImageBackground>
    <ScrollViewMock>
      <View
        style={
          Object {
            "marginLeft": 15,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="calendar"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Esdeveniments      
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="mortar-board"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Gestor de Notes     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="desktop"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Disponibilitat Labs     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="book"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Sales Bibl. BRGF     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="calendar-o"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Cerca Aules Lliures     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="eye"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Places Matrícula     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="plane"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Programes de Mobilitat     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="blackboard"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Lectures TFG i TFM     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="external-link"
              size={25}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Enllaços d'Interès     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="bug"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Feedback & Bugs     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="gear"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Configuració     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="log-out"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Log Out     
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </ScrollViewMock>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "bottom": 10,
          "position": "absolute",
          "right": 20,
        }
      }
    >
      v1.5. 
    </Text>
  </View>
</View>
`;

exports[`Testing Drawer component renders as expected when token provided 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <ImageBackground
      source={
        Object {
          "testUri": "../../../static/fondo2.png",
        }
      }
      style={
        Object {
          "height": 150,
          "marginBottom": 20,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 40,
            "elevation": 8,
            "height": 80,
            "margin": 12,
            "marginBottom": 9,
            "marginTop": 18,
            "overflow": "hidden",
            "width": 80,
          }
        }
      >
        <Image
          source={
            Object {
              "headers": Object {
                "Accept-Language": "ca",
                "Authorization": "Bearer aToken",
              },
              "uri": "https://api.fib.upc.edu/v2/jo/foto.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 40,
              "height": 80,
              "resizeMode": "cover",
              "width": 80,
            }
          }
        />
      </View>
      <View />
    </ImageBackground>
    <ScrollViewMock>
      <View
        style={
          Object {
            "marginLeft": 15,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="calendar"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Esdeveniments      
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="mortar-board"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Gestor de Notes     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="desktop"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Disponibilitat Labs     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="book"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Sales Bibl. BRGF     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="calendar-o"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Cerca Aules Lliures     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="eye"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Places Matrícula     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="plane"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Programes de Mobilitat     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="blackboard"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Lectures TFG i TFM     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="external-link"
              size={25}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Enllaços d'Interès     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="bug"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Feedback & Bugs     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="gear"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Configuració     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="log-out"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Log Out     
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </ScrollViewMock>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "bottom": 10,
          "position": "absolute",
          "right": 20,
        }
      }
    >
      v1.5. 
    </Text>
  </View>
</View>
`;

exports[`Testing Drawer component renders user info correctly after fetching user data 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <ImageBackground
      source={
        Object {
          "testUri": "../../../static/fondo2.png",
        }
      }
      style={
        Object {
          "height": 150,
          "marginBottom": 20,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 40,
            "elevation": 8,
            "height": 80,
            "margin": 12,
            "marginBottom": 9,
            "marginTop": 18,
            "overflow": "hidden",
            "width": 80,
          }
        }
      >
        <Image
          source={
            Object {
              "headers": Object {
                "Accept-Language": "ca",
                "Authorization": "Bearer aToken",
              },
              "uri": "https://api.fib.upc.edu/v2/jo/foto.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 40,
              "height": 80,
              "resizeMode": "cover",
              "width": 80,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginLeft": 15,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#505050",
              "fontWeight": "400",
            }
          }
        >
          Joan Marc Pastor Mackenzie       
        </Text>
        <Text>
          joan.marc.pastor@est.fib.upc.edu       
        </Text>
      </View>
    </ImageBackground>
    <ScrollViewMock>
      <View
        style={
          Object {
            "marginLeft": 15,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="calendar"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Esdeveniments      
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="mortar-board"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Gestor de Notes     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="desktop"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Disponibilitat Labs     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="book"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Sales Bibl. BRGF     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="calendar-o"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Cerca Aules Lliures     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="eye"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Places Matrícula     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="plane"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Programes de Mobilitat     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="blackboard"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
               Lectures TFG i TFM     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="external-link"
              size={25}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Enllaços d'Interès     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="bug"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Feedback & Bugs     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="gear"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Configuració     
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#505050"
              name="log-out"
              size={26}
            />
            <Text
              style={
                Object {
                  "color": "#606060",
                  "marginLeft": 10,
                }
              }
            >
              Log Out     
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </ScrollViewMock>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "bottom": 10,
          "position": "absolute",
          "right": 20,
        }
      }
    >
      v1.5. 
    </Text>
  </View>
</View>
`;
